### ISSUE TITLE
Qusetion about benchmarks with cross_validation_with_val_set

### ISSUE THREAD

**Issue by jiaruHithub** [pyg-team/pytorch_geometric#1384]
## ❓ Questions & Help

I use benchmarks [main.py](https://github.com/rusty1s/pytorch_geometric/blob/master/benchmark/kernel/main.py)
I use SAGPool ,but I found the std is so large.
I think the reason is in one cross_validation, the seed of every experiment is different.Am I right?
If I want to make the experiment stabilize,should I use the same seed in one cross_validation's ten experiments?

**Comment by rusty1s** [comment]
The seed is always the same for each experiment, see [here](https://github.com/rusty1s/pytorch_geometric/blob/master/benchmark/kernel/train_eval.py#L83).

**Comment by jiaruHithub** [comment]
> The seed is always the same for each experiment, see [here](https://github.com/rusty1s/pytorch_geometric/blob/master/benchmark/kernel/train_eval.py#L83). A large stddev is to be expected for some of the datasets since most of them are rather small.

Fine, 
But in papers like SAGPool and ASAP, the stddev always lower than 1.
My stddev is larger than 3.I don;t know why .
Do you have advice reduce the stddev?Thank you!

**Comment by rusty1s** [comment]
That can have of course multiple reasons, like different hyperparameters or a different evaluation metric. Most papers do not report cross validation based on a hold-out validation set, but instead report best validation accuracy (where the stddev should be lower).

**Comment by jiaruHithub** [comment]
Fine, maybe I need become a superman of adjust the hyperparameters.
Thank you, friends!

**Comment by jiaruHithub** [comment]
> That can have of course multiple reasons, like different hyperparameters or a different evaluation metric. Most papers do not report cross validation based on a hold-out validation set, but instead report best validation accuracy (where the stddev should be lower).

I'm sorry to reopen the question.
Could you tell me what ```best validation accuracy``` and stddev.
Like I use 3-layers SAGPool and hidden-size is 128.
If I splite the dataset to three partion:train_set, val_set and tes_set,```best validation accuracy``` is highest value of many experiments(not cross validation) on val_set or test_set.(I think the acc of val_set and test_set is almost equivalent).
Am I right?Thank you,friends!

**Comment by rusty1s** [comment]
Many papers apply 10-fold cross-validation (using 9 splits for training) and pick the best hyperparameters across all 10 runs. For example, you choose the best epoch that gets the highest mean validation accuracy. Therefore, you report best validation accuracy, but do not apply it to an unseen test set (like we do in our benchmarking framework).

**Comment by jiaruHithub** [comment]
> Many papers apply 10-fold cross-validation (using 9 splits for training) and pick the best hyperparameters across all 10 runs based on validation/test performance. For example, you choose the best epoch that gets the highest mean validation accuracy. Therefore, these papers report best validation accuracy, but do not apply your best model to an unseen test set (like we do in our benchmarking framework).

Oh, I know!The best hyperparameters was picked from like num_layers and hidden_size.
And I found there are 1000 items in accs(I only use 3-layers SAGPool on DD).If I choose the best epoch that gets the highest mean validation accuracy.The number of accs may only has 100 items.Therefore the stddev must be lower.
Am I right?

**Comment by jiaruHithub** [comment]
> Many papers apply 10-fold cross-validation (using 9 splits for training) and pick the best hyperparameters across all 10 runs based on validation/test performance. For example, you choose the best epoch that gets the highest mean validation accuracy. Therefore, these papers report best validation accuracy, but do not apply your best model to an unseen test set (like we do in our benchmarking framework).

Sorry,after I took a closer look at the code,Ifound finally the ```acc``` for compute ```mean``` and ```stddev``` is only picked the minimum loss and It's the corresponding accuracy.So it only 10 accs for final calculation.
Am I right?friends
